pkg/bootstrap/bootstrap_config.go

```diff
@@ -301,17 +303,24 @@ func WriteBootstrap(config *meshconfig.ProxyConfig, node string, epoch int, pilo
 	opts["cluster"] = config.ServiceCluster
 	opts["nodeID"] = node
 
-	// Populate the platform locality if available.
-	l := platform.GetPlatformLocality()
+	// Support passing extra info from node environment as metadata
+	meta := getNodeMetaData(localEnv)
+
+	localityOverride := model.GetLocalityOrDefault("", meta)
+	l := util.ConvertLocality(localityOverride)
+	if l == nil {
+		// Populate the platform locality if available.
+		l = platform.GetPlatformLocality()
+	}
 	if l.Region != "" {
 		opts["region"] = l.Region
 	}
 	if l.Zone != "" {
 		opts["zone"] = l.Zone
 	}
-
-	// Support passing extra info from node environment as metadata
-	meta := getNodeMetaData(localEnv)
+	if l.SubZone != "" {
+		opts["sub_zone"] = l.Zone
```

Comment:
opts["sub_zone"] = l.SubZone  ?

Reply:
thank you, good catch. I should update the tests to include sub zone